pandas/core/window_/aggregators.py

```diff
@@ -0,0 +1,136 @@
+import abc
+from typing import Optional
+
+import numpy as np
+
+from pandas._typing import Scalar
+
+
+class BaseAggregator(abc.ABC):
+    """Interface to return the current value of the rolling aggregation at the current step"""
+    def __init__(self, values: np.ndarray) -> None:
+        self.values = values
+
+    @abc.abstractmethod
+    def query(
+        self, start: int, stop: int, previous_start: int, previous_end: int
```

Comment:
Discussed this a bit with @cpcloud: since `previous_start` and `previous_end` are only useful for the `SubtractableAggregator`, we should remove it from the arguments to query and push them to the state of the aggregator itself instead. This gives the implementer of custom aggregators freedom over the state they need to compute their windows efficiently.